sklearn/decomposition/tests/test_pca.py

```diff
@@ -219,20 +225,22 @@ def test_pca_check_projection_list(svd_solver):
 
 @pytest.mark.parametrize("svd_solver", ["full", "arpack", "randomized"])
 @pytest.mark.parametrize("whiten", [False, True])
-def test_pca_inverse(svd_solver, whiten):
+def test_pca_inverse(svd_solver, whiten, global_random_seed):
     # Test that the projection of data can be inverted
-    rng = np.random.RandomState(0)
+    rng = np.random.RandomState(global_random_seed)
     n, p = 50, 3
     X = rng.randn(n, p)  # spherical data
     X[:, 1] *= 0.00001  # make middle component relatively small
     X += [5, 4, 3]  # make a large mean
 
     # same check that we can find the original data from the transformed
     # signal (since the data is almost of rank n_components)
-    pca = PCA(n_components=2, svd_solver=svd_solver, whiten=whiten).fit(X)
+    pca = PCA(
+        n_components=2, svd_solver=svd_solver, whiten=whiten, random_state=rng
+    ).fit(X)
     Y = pca.transform(X)
     Y_inverse = pca.inverse_transform(Y)
-    assert_allclose(X, Y_inverse, rtol=5e-6)
+    assert_allclose(X, Y_inverse, rtol=5e-5)
```

Comment:
Assertion with strict `rtol=5e-6` made the test highly seed-sensitive. Before changing the tolerance, 486 of 600 tests failed after including the `global_random_seed` fixture.
In my opinion, `rtol=5e-5` still asserts good recovery of `X`, since we cannot expect perfect recovery even though the middle column of `X` has very low variance.